tensorflow_similarity/indexer.py

```diff
@@ -104,26 +99,15 @@ def __init__(
         Raises:
             ValueError: Invalid search framework or key value store.
         """
-        distance = distance_canonicalizer(distance)
-        self.distance = distance  # needed for save()/load()
-        self.embedding_output = embedding_output
-        self.embedding_size = embedding_size
-
+        super().__init__(distance, embedding_output, embedding_size, evaluator, stat_buffer_size)
         # internal structure naming
         # FIXME support custom objects
-        self.search_type = search
-        self.kv_store_type = kv_store
-        self.evaluator_type = evaluator
-
-        # stats configuration
-        self.stat_buffer_size = stat_buffer_size
-
-        # calibration
-        self.is_calibrated = False
-        self.calibration_metric: ClassificationMetric = F1Score()
-        self.cutpoints: Mapping[str, Mapping[str, float | str]] = {}
-        self.calibration_thresholds: Mapping[str, np.ndarray] = {}
-
+        self.search_type = search if isinstance(search, str) else type(search).__name__
```

Comment:
interesting, although we are not doing the same for Store class, any particular reason

Fixed

Reply:
So I think the bigger issue here is that type(search).__name__ will return the class name like "NMSLibSearch" which won't match any of the checks below in the init_structure. Same goes for the kv_store.

I think this is the same issue with this pattern that we talked about earlier today. Essentially we accept `Search | str` here, and the string can contain typos, is not restricted like an Enum is. Additionally, we now require the objects to store these string identifiers so that we can store them in the configs and use them to create the objects on reload.